src/libraries/System.Text.Json/tests/Serialization/ConstructorTests.Stream.cs

```diff
@@ -2,111 +2,135 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Collections.Generic;
 using System.IO;
 using System.Threading.Tasks;
 using Xunit;
 
 namespace System.Text.Json.Serialization.Tests
 {
-    public static partial class ConstructorTests
+    public abstract partial class ConstructorTests
     {
-        [Theory]
-        [MemberData(nameof(StreamTestData))]
-        public static async Task ReadSimpleObjectAsync(Type type, byte[] testData)
+        [Fact]
+        public void ReadSimpleObjectAsync()
         {
-            using (MemoryStream stream = new MemoryStream(testData))
+            async Task RunTest<T>(byte[] testData)
             {
-                JsonSerializerOptions options = new JsonSerializerOptions
+                using (MemoryStream stream = new MemoryStream(testData))
                 {
-                    DefaultBufferSize = 1
-                };
+                    JsonSerializerOptions options = new JsonSerializerOptions
+                    {
+                        DefaultBufferSize = 1
+                    };
 
-                var obj = await JsonSerializer.DeserializeAsync(stream, type, options);
-                ((ITestClass)obj).Verify();
+                    var obj = await JsonSerializer.DeserializeAsync<T>(stream, options);
+                    ((ITestClass)obj).Verify();
+                }
             }
+
+            // Simple models can be deserialized.
+            Task.Run(async () => await RunTest<Parameterized_IndexViewModel_Immutable>(Parameterized_IndexViewModel_Immutable.s_data));
```

Comment:
Using `Task.Run` like this will "fire and forget" since the task is not awaited anywhere else. This could cause random exceptions later. The inner `await` doesn't change the semantics that AFAIK.

If you want to run all of these async, get the task from each and call `Task.WhenAll` on the set. That will ensure the tasks finish before this test returns.